lib/renderer/init.js

```diff
@@ -141,7 +141,20 @@ if (nodeIntegration === 'true') {
 
   // Set the __filename to the path of html file if it is file: protocol.
   if (window.location.protocol === 'file:') {
-    var pathname = process.platform === 'win32' && window.location.pathname[0] === '/' ? window.location.pathname.substr(1) : window.location.pathname
+    let pathname
+    const loc = window.location
+    if (process.platform === 'win32') {
+      if (loc.pathname[0] === '/') {
+        pathname = loc.pathname.substr(1)
+      }
+
+      if (loc.hostname.length > 0 && globalPaths[0].startsWith('\\')) {
+        pathname = `//${loc.host}/${pathname}`
+      }
+    } else {
+      pathname = loc.pathname
```

Comment:
Can you please initialise `let pathname` with `window.location.pathname` and do all Windows specific stuff below it? Like this:

```js
const location = window.location
let pathname = location.pathname

if (process.platform === 'win32') {
  // Oh crap
  ...
}
```